src/direct/lwt_direct.mli

```diff
@@ -0,0 +1,95 @@
+(** Direct style control flow for Lwt.
+
+    This module relies on OCaml 5's
+    {{:https://ocaml.org/manual/5.3/effects.html} effect handlers}.
+    Instead of chaining promises using {!Lwt.bind} and {!Lwt.map}
+    and other combinators, it becomes possible to start
+    lightweight "tasks" using [Lwt_direct.spawn (fun () -> ...)].
+    The body of such a task is written in direct-style code,
+    using OCaml's standard control flow structures such as loops,
+    higher-order functions, exception handlers, [match], etc.
+
+    Interactions with the rest of lwt can be done using [await],
+    for example:
+
+    {[
+    Lwt_direct.spawn (fun () ->
+      let continue = ref true in
+      while !continue do
+        match Lwt_io.read_line in_channel |> Lwt_direct.await with
+        | exception End_of_file -> continue := false
+        | line ->
+          let uppercase_line = String.uppercase_ascii line in
+          Lwt_io.write_line out_channel uppercase_line |> Lwt_direct.await
+      done)
+    ]}
+
+    This code snippet contains a simple "task" that repeatedly reads
+    a line from a [Lwt_io] channel, uppercases it, and writes the
+    uppercase version to another channel.
+
+    This task is itself a [unit Lwt.t], which is resolved when the function
+    returns. It is possible to use
+    {!Lwt_direct.run_in_the_background} to ignore the result and
+    let the task run in the background instead.
+
+    *)
+
+val spawn : (unit -> 'a) -> 'a Lwt.t
```

Comment:
I'm not sure `spawn` is a very understandable name. Maybe `promise` (like javascript, …) or `async` (like rust's tokio or miou) would be a bit better?